src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2109,12 +2109,18 @@ private unsafe ParsingError PrivateParseMinimal()
                             _flags |= Flags.UncPath;
                             idx = i;
                         }
+                        else if (!IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && i - idx == 3 && pUriString[i - 1] == '/')
+                        {
```

Comment:
This syntax is the windows UNC syntax, it has no meaning on Unix. There is no reason to add backslash support to it.
I am suggesting to update the pr so file://localhost/ is the same as file:///. It would no longer be a UNC uri, it's LocalPath would be /.